File: CovarianzaYcontraVarianza_C/Scooter.cc
#include "Scooter.h"
#include<iostream>
#include<string>
using namespace std;

Scooter::Scooter(string p,int r,int a):Bicicleta(p,r){
  antiguedad = a;
}

void Scooter::avanzar(){
  cout<<"El Scooter de "<<propietario<<" con ruedas de tamanio "<<tamRuedas<<"cm."
  <<"de antiguedad de "<<antiguedad<<" año(s) Avanza!"<<endl;
}

File: Covarianza y contravarianza java/src/to/Scooter.java
package to;

public class Scooter extends Bicicleta implements Rueda{
	public Scooter(){}
    private int antiguedad;
    
	public Scooter(int antiguedad) {
		super();
		this.antiguedad = antiguedad;
	}
	
	@Override
	public void avanzar() {
		System.out.println("El Scooter avanza!");
	}

	@Override
	public void detener() {
		System.out.println("El Scooter se detuvo!");	
	}
	
}
